docs/master/federation/entity-representations.md

```diff
@@ -0,0 +1,84 @@
+# Entity representation
+
+To reference an entity that originates in another subgraph, first subgraph needs to define a stub of that entity to make
+its own schema valid. The stub includes just enough information for the subgraph to know how to uniquely identify a
+particular entity in another subgraph.
+Read more about the entity representation in the [Apollo Federation docs](https://www.apollographql.com/docs/federation/v1/entities/#entity-representations).
+
+A representation always consists of:
+- A `__typename` field;
+- Values for the entity's primary key fields.
+
+In this way, the resolver should return an array consisting of the `__typename` field, which corresponds to the entity's
+name, and the primary key that can identify the entity.
+
+## Example 1
+
+In this example, subgraph for order service has an entity called `Order`, which in turn has an entity called `Receipt`
+defined in a separate subgraph for payment service. The relationship between `Order` and `Receipt` is one-to-one.
+
+```graphql
+type Order {
+    id: Int!
+    receipt: Receipt! @field(resolver: "App\\GraphQL\\Resolvers\\Receipt")
+}
+
+type Receipt @key(fields: "uuid") @extends{
+    uuid: String! @external
+}
+```
+
+The resolver for receipt in order service returns an array consisting of:
+- `__typename` - the entity name from the payment service;
+- `uuid` - the primary key of the receipt.
+
+```php
+namespace App\GraphQL\Resolvers;
+
+final class Receipt
+{
+    public function __invoke($model): array
+    {
+        return [
+            '__typename' => 'Receipt',
+            'uuid' => $model->receipt_id,
+        ];
+    }
+}
+```
+
+## Example 2
+
+In this example, subgraph for order service has an entity called `Order`, which in turn has an entity called `Product`
+defined in a separate subgraph for product service. The relationship between `Order` and `Product` is one-to-many.
+
+```graphql
+type Order {
+    sum: Int!
+    products: [Product!]! @field(resolver: "App\\GraphQL\\Resolvers\\Products")
+}
+
+type Product @key(fields: "uuid") @extends{
+    uuid: String! @external
```

Comment:
I think for this, no custom resolver would really be necessary if defined like this:

```suggestion
    products: [Product!]! @hasMany
}

type Product @key(fields: "id") @extends{
    id: String! @external
```

Reply:
In this case, it is necessary to define the `products` relationship in Order model. Perhaps it's worth keeping the current example for the case when there is no relationship and adding a new example for the one you described?

Reply:
I would not mind having an example for a non-Eloquent non-relation use-case, but I think it is misleading to have a custom resolver for Eloquent relations. It is probably a good idea to have both to show both approaches and the appropriate implementation for each.

Reply:
Alright, I will provide examples for both cases, I believe it will indeed be helpful.